editoast/src/views/work_schedules.rs

```diff
@@ -275,6 +295,238 @@ async fn project_path(
+    responses(
+        (status = 200, body = WorkScheduleGroupCreateResponse, description = "The id of the created work schedule group"),
+    )
+)]
+async fn create_group(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    Json(WorkScheduleGroupCreateForm {
+        work_schedule_group_name,
+    }): Json<WorkScheduleGroupCreateForm>,
+) -> Result<Json<WorkScheduleGroupCreateResponse>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let group_name = work_schedule_group_name.unwrap_or(Uuid::new_v4().to_string());
+
+    // Create the work_schedule_group
+    let work_schedule_group = WorkScheduleGroup::changeset()
+        .name(group_name.clone())
+        .creation_date(Utc::now())
+        .create(conn)
+        .await;
+    let work_schedule_group = work_schedule_group.map_err(|e| map_diesel_error(e, group_name))?;
+    Ok(Json(WorkScheduleGroupCreateResponse {
+        work_schedule_group_id: work_schedule_group.id,
+    }))
+}
+
+#[utoipa::path(
+    delete, path = "",
+    tag = "work_schedules",
+    params(WorkScheduleGroupIdParam),
+    responses(
+        (status = 204, description = "The work schedule group has been deleted"),
+        (status = 404, description = "The work schedule group does not exist"),
+    )
+)]
+async fn delete_group(
+    State(app_state): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    raw_group_id: Path<WorkScheduleGroupIdParam>,
+) -> Result<impl IntoResponse> {
+    let authorized = auth
+        .check_roles([BuiltinRole::WorkScheduleWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let group_id = raw_group_id.id;
+
+    conn.transaction(|conn| {
+        Box::pin(async move {
+            // Check that the group exists
+            let work_schedule_group =
+                WorkScheduleGroup::retrieve_or_fail(&mut conn.clone(), group_id, || {
+                    WorkScheduleError::WorkScheduleGroupNotFound { id: group_id }
+                })
+                .await?;
+
+            // List work schedules to be deleted
+            let selection_setting = SelectionSettings::new()
+                .filter(move || WorkSchedule::WORK_SCHEDULE_GROUP_ID.eq(group_id));
+            let work_schedule_ids = WorkSchedule::list(&mut conn.clone(), selection_setting)
+                .await?
+                .into_iter()
+                .map(|work_schedule| work_schedule.id);
+
+            // Delete them
+            use crate::models::DeleteBatch;
```

Comment:
Shouldn't it be with the other imports at the top of the file even though it is only being used there ?

Reply:
Right, I copied what was [there](https://github.com/OpenRailAssociation/osrd/blob/dev/editoast/src/views/train_schedule.rs#L256)

Fixed in a1bd117b3c855cecd04c1dee1aba55d9435fcdd9